temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/WorkersAutoDiscoveryAutoConfiguration.java

```diff
@@ -0,0 +1,114 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.spring.boot.autoconfigure;
+
+import io.temporal.spring.boot.ActivityImpl;
+import io.temporal.spring.boot.WorkflowImpl;
+import io.temporal.spring.boot.autoconfigure.properties.TemporalProperties;
+import io.temporal.worker.Worker;
+import io.temporal.worker.WorkerFactory;
+import java.util.*;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.aop.support.AopUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.support.BeanDefinitionValidationException;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.context.annotation.*;
+import org.springframework.core.annotation.AnnotationUtils;
+import org.springframework.core.type.filter.AnnotationTypeFilter;
+
+/**
+ * Works only if `spring.temporal.workersAutoDiscovery` is true and responsible for auto-discovery
+ * of workflow and activity implementations that are annotated with {@link
+ * io.temporal.spring.boot.ActivityImpl} and {@link io.temporal.spring.boot.WorkflowImpl}
+ */
+@Configuration
+@EnableConfigurationProperties(TemporalProperties.class)
+@Import({WorkersAutoDiscoveryAutoConfiguration.ActivityAutoDiscoveryProcessor.class})
+@Conditional(WorkersAutoDiscoveryPackagesPresentCondition.class)
+public class WorkersAutoDiscoveryAutoConfiguration {
+  private static final Logger log =
+      LoggerFactory.getLogger(WorkersAutoDiscoveryAutoConfiguration.class);
+
+  private final TemporalProperties properties;
+
+  public WorkersAutoDiscoveryAutoConfiguration(TemporalProperties properties) {
+    this.properties = properties;
+  }
+
+  @Bean(name = "autoDiscoveredWorkflowImplementations")
+  public Collection<Class<?>> autoDiscoveredWorkflowImplementations() {
+    ClassPathScanningCandidateComponentProvider scanner =
+        new ClassPathScanningCandidateComponentProvider(false);
+    scanner.addIncludeFilter(new AnnotationTypeFilter(WorkflowImpl.class));
+    Set<Class<?>> implementations = new HashSet<>();
+    for (String pckg : properties.getWorkersAutoDiscovery().getPackages()) {
+      Set<BeanDefinition> candidateComponents = scanner.findCandidateComponents(pckg);
+      for (BeanDefinition beanDefinition : candidateComponents) {
+        try {
+          implementations.add(Class.forName(beanDefinition.getBeanClassName()));
+        } catch (ClassNotFoundException e) {
+          throw new BeanDefinitionValidationException(
+              "Fail loading class for bean definition " + beanDefinition, e);
+        }
+      }
+    }
+    return implementations;
```

Comment:
I wonder if you want to find name + task queue clashes here or let it fail at worker validation time.

Reply:
It will fail on registration on the workers. I will try to not bring logic like this into spring boot as long as possible. What can be shared, should be shared. The only reason why it can leak here is if we want to give some super-specific error message that can reference a spring boot config.